### DISCUSSION TITLE
Experiences, intuition, and results from training DALL-E models

### DISCUSSION THREAD

**Discussion by TheodoreGalanos** [lucidrains/DALLE-pytorch#95] [General]
Hello everyone!

Now that the codebase is getting more stable I was wondering if this is a good place to discuss ideas about how to actually train a DALL-E model (outside of using pretrained models). Questions like

* What kind of applications do you want to train a dall-e model for? How are you creating your datasets?
* What parameters seem to work? Is there any combination that does? How long does it need to be trained
* What kind of annotations did you go for and did they work alright? Do multiple annotations help or limit model training?

I'm currently trying to see if all this works on a small, narrow dataset (5000 image/text pairs of building layouts, mostly residential). I'm using simple annotations like "A house with 3 rooms and 2 bathrooms" and I added multiple annotations for each image in the form of adjacencies, e.g. "A house where the living room is adjacent to the kitchen but not adjacent to the bathroom". It's arguably a bit nutty and I might have overcomplicated things. I'll know pretty soon and share here.

The next experiment I'm running is urban design. I'll pass thousands of urban areas through a surrogate wind comfort model and annotate each of them according to performance, e.g. "A dense urban design with large areas of wind comfort" (or smth smarter). I'm hoping the urban setting is more diverse and allows for some quirkier results.

Curious to hear your stories about this!
Regards,
Theodore.

**Comment by afiaka87** [comment]
@TheodoreGalanos 

I've had decent luck with "small" datasets! Nothing quite as small as yours though, so do let me know if you have luck there. 


In general, the "million scraped image-text pairs from open ai" data set looked like it was _probably_ overfitting? But nevertheless, you could fairly easily take that checkpoint (seriously, it's on wandb.ai if you want it) and have a reasonable "mannequin generator". The reason it worked so well on the mannequins was that every single image of the mannequins was in an identical format (the caption also _always_ had the same text except for the change in color and texture of the shirt and pants). 

It's probably important to keep in mind that this dataset specifically might be a little too likely to make CLIP happy, considering it was generated (partially) with the very same or similar pretrained CLIP given to us by OpenAI.

Having said that, I think there's actually high hopes for "domain specific DALL-E's" that only know about (for instance) fashion. If, however, we find that a humongous dataset can generalize that information as well, then I suppose I would be wrong about that. Time will tell.
